metabase/metabase
# Clicking outside of the Actions Editor loses your work

Closes https://github.com/metabase/metabase/issues/30539

### Description

Currently, when editing actions, any click outside the modal discards their changes without warning and forces the user to start again. To quote the issue, "that sucks."

I've added an option to the Modal component that allows us to choose whether we want to enable the `onClickOutside` wrapper, and I've added that to Action modals. We can also use this flag for other modals in the future if we need.

### How to verify

_Testing Action Modal_
1. Ensure you have actions enabled for the Sample Database (or whatever database you desire)
2. Create a new Action by clicking `+ New` in the top right corner
3. Try clicking outside the modal. It shouldn't close.
4. Click 'Cancel' - the modal should now close.

_Test that other modals behave the same way - ex: Save Modal_
1. Navigate to a collection with any sort of item in it
5. Go to options and click `Move`
6. You should be able to click outside the modal and it will close.

### Checklist



- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/metabase/metabase/30697)
